cpp/oneapi/dal/backend/math.hpp

```diff
@@ -0,0 +1,26 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+namespace oneapi::dal::backend {
+
+template <typename float_t>
+inline constexpr float_t exp_treshold() {
+    return std::is_same_v<float_t, double> ? -650.0 : -75.0f;
```

Comment:
It's completely unclear for me why:
1. This is a common threshold for the whole library
2. Why it has such values

Reply:
1) This function is used in different algorithms in DAAL interfaces and MKL:
- https://github.com/oneapi-src/oneDAL/blob/e5864a1c449b8f45e8def5986fe26abcea20bda4/cpp/daal/src/sycl/math_service_types.h#L40
- https://github.com/oneapi-src/oneDAL/blob/e5864a1c449b8f45e8def5986fe26abcea20bda4/cpp/daal/src/externals/service_math_mkl.h#L212
- https://github.com/oneapi-src/oneDAL/blob/e5864a1c449b8f45e8def5986fe26abcea20bda4/cpp/daal/src/externals/service_math_mkl.h#L294
2) minimal float value ~ 1.2e-38, exp(-75) ~ 2.6e-33
minimal double value ~ 2.3e-308, exp(-650) ~ 5.1e-283

Reply:
Can you please place explanations from p.2 to the file as comments?

Reply:
And don't use magic constants, figure out meaningful names for thresholds.